sw/device/tests/rstmgr_alert_info_test.c

```diff
@@ -0,0 +1,685 @@
+static dif_rstmgr_t rstmgr;
+static dif_alert_handler_t alert_handler;
+static dif_uart_t uart0, uart1, uart2, uart3;
+static dif_otp_ctrl_t otp_ctrl;
+static dif_spi_host_t spi_host;
+static dif_rv_plic_t plic;
+static dif_aon_timer_t aon_timer;
+static dif_pwrmgr_t pwrmgr;
+static dif_i2c_t i2c0, i2c1, i2c2;
+
+#define kNumClasses kDifAlertHandlerClassD + 1
+#define kNumAlerts kTopEarlgreyAlertIdLast + 1
+
+static uint64_t merge32(uint32_t x, uint32_t y) {
+  uint64_t pow = 10;
+  while (y >= pow) {
+    pow *= 10;
+  }
+  return (uint64_t)(x * pow + y);
+}
+
+typedef struct node {
+  const char *name;
+  dif_alert_handler_alert_t alert;
+  dif_alert_handler_class_t class;
+} node_t;
+
+typedef enum cstate {
+  kCstateIdle = 0,
+  kCstateTimeout = 1,
+  kCstateTerminal = 3,
+  kCstatePhase0 = 4,
+  kCstatePhase1 = 5,
+  kCstatePhase2 = 6,
+  kCstatePhase3 = 7,
+  kCstateFsmError = 2
+} cstate_t;
+
+typedef enum test_round {
+  kRound1 = 0,
+  kRound2,
+  kRound3,
+  kRoundTotal
+} test_round_t;
+
+typedef struct alert_info {
+  char *test_name;
+  uint64_t alert_cause;                   // 61bit
+  uint8_t loc_alert_cause;                // 7bit
+  uint16_t class_accum_cnt[kNumClasses];  // 4x16bit
+  uint32_t class_esc_cnt[kNumClasses];    // 4x32bit
+  cstate_t class_esc_state[kNumClasses];  // 4x3bit
+} alert_info_t;
+
+static test_round_t global_test_round;
+static uint32_t global_alert_called;
+static dif_alert_handler_escalation_phase_t esc_profiles[][kNumClasses] = {
+    [kDifAlertHandlerClassA] = {{.phase = kDifAlertHandlerClassStatePhase0,
+                                 .signal = 0,
+                                 .duration_cycles = 5000},
+                                {.phase = kDifAlertHandlerClassStatePhase1,
+                                 .signal = 0,
+                                 .duration_cycles = 3000}},
+    [kDifAlertHandlerClassB] = {{.phase = kDifAlertHandlerClassStatePhase1,
+                                 .signal = 0,
+                                 .duration_cycles = 3000}},
+    [kDifAlertHandlerClassC] = {{.phase = kDifAlertHandlerClassStatePhase0,
+                                 .signal = 0,
+                                 .duration_cycles = 5000},
+                                {.phase = kDifAlertHandlerClassStatePhase1,
+                                 .signal = 0,
+                                 .duration_cycles = 3000}},
+    [kDifAlertHandlerClassD] = {{.phase = kDifAlertHandlerClassStatePhase0,
+                                 .signal = 0,
+                                 .duration_cycles = 7200},
+                                {.phase = kDifAlertHandlerClassStatePhase1,
+                                 .signal = 1,
+                                 .duration_cycles = 4800},
+                                {.phase = kDifAlertHandlerClassStatePhase2,
+                                 .signal = 3,
+                                 .duration_cycles = 2400}}};
```

Comment:
Could you briefly explain why these alert phase cycles are different?

Reply:
I assume each class profile is independent. 
In each classes, I put some arbitrary numbers for the duration cycles.
Signals are not arbitrary, though. I intentionally avoid to going signal 1 except
class D, because I don't want to trigger escalation reset in other classes but class D.